buildSrc/src/main/kotlin/org/gradle/plugins/ideconfiguration/IdeConfigurationPlugin.kt

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugins.ideconfiguration
+
+import org.gradle.api.*
+import org.gradle.api.Project
+import org.gradle.api.plugins.BasePluginConvention
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.tasks.Copy
+import org.gradle.plugins.ide.eclipse.model.EclipseModel
+import org.gradle.kotlin.dsl.*
+import org.gradle.plugins.ide.eclipse.model.AbstractClasspathEntry
+import org.gradle.plugins.ide.eclipse.model.Classpath
+import org.gradle.plugins.ide.eclipse.model.SourceFolder
+import org.gradle.plugins.ide.idea.model.*
+import org.gradle.plugins.pegdown.PegDown
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Entities
+import org.jsoup.parser.Parser
+import java.io.File
+
+internal
+const val ideConfigurationBaseName = "ideConfiguration"
+
+
+class IdeConfigurationPlugin : Plugin<Project> {
```

Comment:
:x: plugin class should be `open`

Reply:
Which derived classes should be open would be good to document.